src/main/java/com/uber/cadence/samples/hello/HelloCancellation.java

```diff
@@ -0,0 +1,149 @@
+ */
+
+package com.uber.cadence.samples.hello;
+
+import static com.uber.cadence.samples.common.SampleConstants.DOMAIN;
+
+import com.uber.cadence.activity.ActivityMethod;
+import com.uber.cadence.client.WorkflowClient;
+import com.uber.cadence.client.WorkflowOptions;
+import com.uber.cadence.client.WorkflowStub;
+import com.uber.cadence.worker.Worker;
+import com.uber.cadence.workflow.Workflow;
+import com.uber.cadence.workflow.WorkflowMethod;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+
+/**
+ * Demonstrates triggering an activity in response to a cancellation request. Requires a local
+ * instance of Cadence server to be running.
+ */
+public class HelloCancellation {
+  static final String TASK_LIST = "HelloCancellation";
+
+  /** Workflow interface has to have at least one method annotated with @WorkflowMethod. */
+  public interface GreetingWorkflow {
+    /** @return greeting string. If cancelled then it will invoke the sayGoodbye activity. */
+    @WorkflowMethod(executionStartToCloseTimeoutSeconds = 20, taskList = TASK_LIST)
+    String getGreeting(String name) throws Exception;
+  }
+
+  /** Activity interface is just a POJI. */
+  public interface GreetingActivities {
+    @ActivityMethod(scheduleToCloseTimeoutSeconds = 2)
+    String composeGreeting(String greeting, String name) throws Exception;
+
+    @ActivityMethod(scheduleToCloseTimeoutSeconds = 2)
+    String sayGoodbye(String name);
+  }
+
+  /** GreetingWorkflow implementation that calls GreetingsActivities#composeGreeting. */
+  public static class GreetingWorkflowImpl implements GreetingWorkflow {
+
+    /**
+     * Activity stub implements activity interface and proxies calls to it to Cadence activity
+     * invocations. Because activities are reentrant, only a single stub can be used for multiple
+     * activity invocations.
+     */
+    private final GreetingActivities activities =
+        Workflow.newActivityStub(GreetingActivities.class);
+
+    @Override
+    public String getGreeting(String name) throws Exception {
+      try {
+        Workflow.sleep(Duration.ofSeconds(10));
+        return activities.composeGreeting("Hello", name);
+      } catch (
+          CancellationException
+              e) { // This exception is thrown when a cancellation is requested on the current
+        // workflow
+        /**
+         * Any call to an activity or a child workflow after the workflow is cancelled is going to
+         * fail immediately with the CancellationException. the DetachedCancellationScope doesn't
+         * inherit its cancellation status from the enclosing scope. Thus it allows running a
+         * cleanup activity even if the workflow cancellation was requested.
+         */
+        Workflow.newDetachedCancellationScope(() -> activities.sayGoodbye(name));
+      }
+      return "Cancellation issued.. Goodbye.";
```

Comment:
return "ignored due to cancellation"

Reply:
removed return statement and threw the exception